### DISCUSSION TITLE
AppStoreConnect Introductory Pricing Support

### DISCUSSION THREAD

**Discussion by pfay-fc** [fastlane/fastlane#16818] [Ideas]
Feature Requests should be in discussions now, re-requesting / moving this one: https://github.com/fastlane/fastlane/issues/13089

In late 2017 apple added Introductory Offers and I don't believe it has been implemented in fastlane. I was somewhat surprised that it doesn't seem to be integrated so please correct me if I'm wrong.

### Feature Request
Support for accessing (read/write) Introductory Pricing ("introOffer" of raw_pricing_data) for subscriptions.

![image](https://user-images.githubusercontent.com/66968089/87109724-43c2ce80-c21a-11ea-98df-7a3ee538556f.png)

**Motivation Behind Feature**
Otherwise manual steps are required to manage subscriptions with introductory offers. Many subscriptions will want to offer free trials or other such introductory offers.

**Feature Description**
Read/write access to manage introductory offers. Either in details, or at least the ability to set some period as a free trial. (Maybe carry forward support for the `subscription_free_trial` field)

**Alternatives or Workarounds**
Manually manage introductory pricing via App Store Connect

**Related Issues**
https://github.com/fastlane/fastlane/issues/13991 has some more details, more from the perspective of `subscription_free_trial` no longer being supported
https://github.com/fastlane/fastlane/issues/11189 is another related issue

This PR https://github.com/fastlane/fastlane/pull/11627 fixed the ability to set pricing for renewables but (intentionally) did not go so far as to add introductory pricing support.

App Store Connect also has a similar feature called Promotional Offers which was added later (early 2019?), but I think that's a separate feature request that I'm not personally invested in currently and I suspect is less commonly used: https://developer.apple.com/app-store/subscriptions/#subscription-offers There may be some implementation similarities with that feature, I haven't looked into it though.

**Comment by joshdholtz** [comment]
@pfay-fc Thanks for putting this in discussions! I’ve been refreshing all evening waiting for new things to appear ❤️

I do think this should totally be in fastlane 💪 I don’t think I was aware this was a thing but I can totally look into what APIs it uses and implement them.

I have a bit of cleanup I still need to do from the most recent set of App Store Connect changes but we can for sure get this added in 🙂

**Reply by joshdholtz** [reply]
And my apologies in the first issue getting lost! I’m really hoping discussions allows features requests to stand out and get implemented (or implemented faster) 🤞

**Comment by schunlee** [comment]
`require 'spaceship'`

`Spaceship::Tunes.login($FASTLANE_USER, $FASTLANE_PASSWORD)`
`app = Spaceship::Application.find(ENV['APP_IDENTIFIER'])`

`purch_id = ""`
`intro = nil`
`app.in_app_purchases.all.each do | purch |`
`    if purch.product_id == $PRODUCT_ID`
`        intro = purch.edit.raw_pricing_data["introOffers"]`
`        purch_id = purch.edit.purchase_id`
`    end`
`end`

`puts intro`
`puts purch_id`

with into and purch_id, using 'faraday' to execute a POST request, I think can create an Introductory Pricing.
https://appstoreconnect.apple.com/WebObjects/iTunesConnect.woa/ra/apps/iaps/pricing/batch  POST
